engines/dfc_provider/spec/serializers/dfc_provider/catalog_item_serializer_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe DfcProvider::CatalogItemSerializer do
+  let!(:product) { create(:simple_product ) }
+  let!(:variant) { product.variants.first }
+
+  subject { described_class.new(variant) }
+
+  describe '#id' do
+    it 'returns the expected value' do
+      expect(subject.id).to eq(
+        DfcProvider::Engine.routes.url_helpers.api_dfc_provider_enterprise_catalog_item_url(
+          enterprise_id: product.supplier_id,
+          id: variant.id,
+          host: 'http://test.host'
+        )
+      )
```

Comment:
It's often better to test for the computer result because it gives the reader the opportunity to see what's really returned and it verifies that the URL generator is also correct. If both sides, the code and the test, use a faulty URL generator, the test will pass and nobody notices.